src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/HardcodedSymmetricAlgorithmKeysSources.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    internal static class HardcodedSymmetricAlgorithmKeysSources
+    {
+        /// <summary>
+        /// <see cref="SourceInfo"/>s for hardcoded symmetric algorithm cryptographic keys tainted data sources.
+        /// </summary>
+        public static ImmutableHashSet<SourceInfo> SourceInfos { get; }
+
+        /// <summary>
+        /// Statically constructs.
+        /// </summary>
+        static HardcodedSymmetricAlgorithmKeysSources()
+        {
+            var builder = PooledHashSet<SourceInfo>.GetInstance();
+
+            builder.AddSourceInfo(
+                WellKnownTypeNames.SystemConvert,
+                isInterface: false,
+                taintedProperties: null,
+                taintedMethodsNeedsPointsToAnalysis: null,
+                taintedMethodsNeedsValueContentAnalysis: new (MethodMatcher, ValueContentCheck[])[]{
+                    (
+                        (methodName, arguments) =>
+                            methodName == "FromBase64String",
+                        new ValueContentCheck[]{
+                            (argumentPointsTos, argumentValueContents) =>
+                            {
+                                return argumentValueContents.Length == 1
+                                    && argumentValueContents[0].LiteralValues.Any(
+                                        (object? v) => v is string s && s.Length % 4 == 0 && IsLegalKeySize(s.Length * 3 / 4));
+                            }
+                        }
+                    ),
+                });
+
+            // Consider also checking the resulting lengths of Encoding.GetBytes() with IsLegalKeySize(). ValueContentCheck
+            // would need to be aware of concrete type.
+            builder.AddSourceInfoSpecifyingTaintedTargets(
+                WellKnownTypeNames.SystemTextEncoding,
+                isInterface: false,
+                taintedProperties: null,
+                taintedMethodsNeedsPointsToAnalysis: null,
+                taintedMethodsNeedsValueContentAnalysis: new (MethodMatcher, (ValueContentCheck, string)[])[]{
+                    (
+                        (methodName, arguments) =>
+                            methodName == "GetBytes" &&
+                            arguments.Count() == 5 &&
```

Comment:
`arguments` is an `ImmutableArray` so you could avoid the method call:

```suggestion
                            arguments.Length == 5 &&
```